app/src/main/java/org/mozilla/fenix/components/metrics/UsageRecorder.kt

```diff
@@ -0,0 +1,44 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.components.metrics
+
+import android.app.Activity
+import android.content.Context
+import mozilla.components.support.utils.ext.getPackageInfoCompat
+import org.mozilla.fenix.android.DefaultActivityLifecycleCallbacks
+import org.mozilla.fenix.utils.Settings
+
+/**
+ * This will store app usage time to disk, based on Resume and Pause lifecycle events. Currently,
+ * there is only interest in usage during the first day after install.
+ */
+class UsageRecorder(private val settings: Settings) : DefaultActivityLifecycleCallbacks {
+    private val activityStartTimes: MutableMap<String, Long?> = mutableMapOf()
+    private val dayMillis: Long = 1000 * 60 * 60 * 24
+
+    override fun onActivityResumed(activity: Activity) {
+        super.onActivityResumed(activity)
+        activityStartTimes[activity.componentName.toString()] = System.currentTimeMillis()
+    }
+
+    override fun onActivityPaused(activity: Activity) {
+        super.onActivityPaused(activity)
+        val startTime = activityStartTimes[activity.componentName.toString()] ?: return
+        val elapsedTimeMillis = System.currentTimeMillis() - startTime
+        settings.usageTimeGrowthData = elapsedTimeMillis + settings.usageTimeGrowthData
+    }
+
+    /**
+     * Whether these lifecycle callbacks should be registered. They will not be registered if growth
+     * data about usage has already been sent, or if it has been longer than a day since the app was
+     * installed since there is no need to store usage data after that period.
+     */
+    fun shouldRegister(context: Context): Boolean {
+        val installTime = context.packageManager
+            .getPackageInfoCompat(context.packageName, 0)
+            .firstInstallTime
+        return !settings.usageTimeGrowthSent && System.currentTimeMillis() < installTime + dayMillis
```

Comment:
some clarifying parameters here might be useful? e.g.
```suggestion
        return !settings.usageTimeGrowthSent && (System.currentTimeMillis() < (installTime + dayMillis))
```